### PR TITLE
Updated Action Hooks and Classes for Wordpress Coding Standards

### PR DESCRIPTION
**Pull request by josh-rathke**
Also fixed a couple “yoda condition” issues to make unit testing pass.

### CODE REVIEW

File: library/offcanvas-walker.php [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
-if ( ! class_exists( 'FoundationPress_offcanvas_walker' ) ) :
-class FoundationPress_offcanvas_walker extends Walker_Nav_Menu {
+if ( ! class_exists( 'FoundationPressOffcanvasWalker' ) ) :
+class FoundationPressOffcanvasWalker extends Walker_Nav_Menu {
 
 	function display_element( $element, &$children_elements, $max_depth, $depth = 0, $args, &$output ) {
 		$element->has_children = ! empty( $children_elements[ $element->ID ] );
 		$element->classes[] = ( $element->current || $element->current_item_ancestor ) ? 'active' : '';
-		$element->classes[] = ( $element->has_children && $max_depth !== 1 ) ? 'has-submenu' : '';
+		$element->classes[] = ( 1 !== $element->has_children && $max_depth ) ? 'has-submenu' : '';
```
Comment (Aetles) [inline library/offcanvas-walker.php]: This change seems wrong, previously it was 
`$element->has_children && $max_depth !== 1` 
and now it is 
`1 !== $element->has_children && $max_depth`
shouldn't it rather be
 `$element->has_children && 1 !== $max_depth `
?

**Comment by Aetles** [comment]
Also, and I feel like I am nitpicking here, but if we are to follow coding standards then The WordPress [Naming Conventions](https://make.wordpress.org/core/handbook/coding-standards/php/#naming-conventions) say this about naming classes:
> Class names should use capitalized words separated by underscores.

**Comment by josh-rathke** [comment]
@Aetles you were totally right. I completely missed that in the menu walkers. It had actually happened in menu-walker.php as well, but I went ahead and changed it. It now reads:
```php
$element->has_children && 1 !== $max_depth
```
Also thanks for reminding me of the true WP Standards definition for classes. I don't know why I was under the impression that it Classes were to be camel-case. Either way, it should all be changed now. I think I hit all the classes, but might need to have a double check on that.

The only thing that remains in question in my mind would be the text domain. WordPress doesn't seem to have a standard for the text-domain. However in the I18n Documentation found [here](https://codex.wordpress.org/I18n_for_WordPress_Developers) it does state the following under the text domain section:
*If you're translating a plugin or a theme, you'll need to use a text domain to denote all text belonging to that plugin. This increases portability and plays better with already existing WordPress tools. The text domain must match the “slug” of the plugin.*

**Comment by Aetles** [comment]
Great! About the text domain: I'd vote for changing `FoundationPress` to `foundationpress` as that is more "slug-like".

**Comment by josh-rathke** [comment]
@olefredrik What do you think about the text domain change?

**Comment by olefredrik** [comment]
I agree that it could be okay to change the text-domain to ```foundationpress```, as long as this change does not affect any existing translations (I see no reason why it should). In all cases, the FoundationPress repository is now build passing, due to your great contributions. Well done, guys! You are awesome!

**Pull request merged**